### ISSUE TITLE
Image adapters break with pillow==9.1.0 for grayscale for bentoml==0.13.1

### ISSUE THREAD

**Issue by gregd33** [bentoml/BentoML#2416]
**Describe the bug**

If you use an `ImageInput` or `AnnotatedImageInput` when `pillow==9.1.0` is installed and try to submit a grayscale image it will error.

The issue is caused by `imageio.imread()` if we pin `imageio==2.9.0`. Unfortunately upgrading `imageio` is not possible since it breaks  bentoml in a different way.

This is *not* an issue for bentoml 1.0 since it uses `Pillow` directly and you'll just get an image with two dimensions that you convert. But since it's not out yet, this might be helpful for others. 

**To Reproduce**

Create a simple bentoml app that uses Image input such as:

```
import bentoml
from bentoml.adapters import JsonOutput, ImageInput

class ImageTest(bentoml.BentoService):

    @bentoml.api(input=ImageInput(), output=JsonOutput())
    def predict(self, image):
        return {
            "size" : str(image.shape)
        }
```

Submit a grayscale image and it will break if `pillow==9.1.0`

**Solution**

For any services that use `bentoml<1.0` you must pin `pillow<9.1.0` and `imageio<=2.9.0`

**Comment by yubozhao** [comment]
Hey @gregd33 Did you get a chance to try out the 1.0 rc version?

**Comment by aarnphm** [comment]
cc @gregd33, I encourage you to try out 1.0 releases as we have a lot of changes comparing to 0.13. In the meantime, the progress of this issue should be addressed via #2753. Feel free to reopen it anytime. 

In this case, I believe a solution is to create warning for this. It is probably sufficient enough.
